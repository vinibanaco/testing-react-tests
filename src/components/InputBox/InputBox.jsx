import React, { useState } from 'react';
import PropTypes from 'prop-types';

function InputBox({ callback }) {
  const [text, setText] = useState('');

  function handleTextChange(e) {
    const { value } = e.target;
    setText(value);
  }

  function handleSubmit(e) {
    e.preventDefault();

    return callback(text);
  }

  return (
    <form action="#" onSubmit={handleSubmit}>
      <input
        type="text"
        name="text"
        onChange={handleTextChange}
        required
        style={{
          border: '1px solid #999',
          borderRadius: '4px 0 0 4px',
          fontSize: '16px',
          padding: '7px',
        }}
      />
      <button
        type="submit"
        style={{
          backgroundColor: '#e0e0e0',
          border: '1px solid #999',
          borderRadius: '0 4px 4px 0',
          fontSize: '14px',
          /**
           * 6px de padding desejado + 2 * 1px para
           * compensar o tamanho da fonte
           */
          padding: '8px 14px',
          marginLeft: '-1px',
        }}
      >
        Enviar
      </button>
    </form>
  );
}

export default InputBox;

InputBox.propTypes = {
  callback: PropTypes.func.isRequired,
};
