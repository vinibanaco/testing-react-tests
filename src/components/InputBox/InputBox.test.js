import React from 'react';
import { shallow } from 'enzyme';

import InputBox from './InputBox';

const clickFn = jest.fn();

describe('InputBox', () => {
  // Render properly with props
  it("should render properly with 'callback' prop", () => {
    const component = shallow(<InputBox callback={clickFn} />);

    expect(component).toMatchSnapshot();
  });

  // Call the callback
  it('should call the callback function with the value of the input', () => {
    const event = { preventDefault: jest.fn() };
    const component = shallow(<InputBox callback={clickFn} />);

    component.find('input').simulate('change', { target: { value: 'Vinícius' } });
    component.find('form').simulate('submit', event);

    expect(event.preventDefault).toHaveBeenCalledTimes(1);
    expect(clickFn).toHaveBeenCalledTimes(1);
    expect(clickFn).toHaveBeenCalledWith('Vinícius');
  });
});
