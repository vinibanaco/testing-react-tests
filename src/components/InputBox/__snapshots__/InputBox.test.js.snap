// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox should render properly with 'callback' prop 1`] = `
<form
  action="#"
  onSubmit={[Function]}
>
  <input
    name="text"
    onChange={[Function]}
    required={true}
    style={
      Object {
        "border": "1px solid #999",
        "borderRadius": "4px 0 0 4px",
        "fontSize": "16px",
        "padding": "7px",
      }
    }
    type="text"
  />
  <button
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "border": "1px solid #999",
        "borderRadius": "0 4px 4px 0",
        "fontSize": "14px",
        "marginLeft": "-1px",
        "padding": "8px 14px",
      }
    }
    type="submit"
  >
    Enviar
  </button>
</form>
`;
